modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultApplicationPolicyInstance.java

```diff
@@ -74,31 +77,29 @@ public DefaultApplicationPolicyInstance(Application application, PolicyTemplate
-              .setArtifactName(parametrization.getId())
-              .setConfigurationFiles(getResourcePaths(template.getDescriptor().getConfigResourceFiles()))
-              .setExecutionClassloader(template.getArtifactClassLoader().getClassLoader())
-              .setServiceRepository(serviceRepository)
-              .setClassLoaderRepository(classLoaderRepository)
-              .setArtifactPlugins(artifactPlugins)
-              .setExtensionManagerFactory(new PolicyTemplateExtensionManagerFactory(application, extensionModelLoaderRepository,
-                                                                                    artifactPlugins,
-                                                                                    new DefaultExtensionManagerFactory()));
-
-      artifactBuilder.withServiceConfigurator(customizationService -> customizationService
-          .overrideDefaultServiceImpl(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER,
-                                      application.getMuleContext().getRegistry()
-                                          .lookupObject(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER)));
-
-      try {
-        policyContext = artifactBuilder.build();
-        policyContext.getMuleContext().start();
-      } catch (MuleException e) {
-        throw new IllegalStateException("Cannot create artifact context for the policy instance", e);
-      }
+    ArtifactContextBuilder artifactBuilder =
+        newBuilder().setArtifactType(APP)
+            .setArtifactProperties(new HashMap<>(parametrization.getParameters()))
+            .setArtifactName(parametrization.getId())
+            .setConfigurationFiles(getResourcePaths(template.getDescriptor().getConfigResourceFiles()))
+            .setExecutionClassloader(template.getArtifactClassLoader().getClassLoader())
+            .setServiceRepository(serviceRepository)
+            .setClassLoaderRepository(classLoaderRepository)
+            .setArtifactPlugins(artifactPlugins)
+            .setExtensionManagerFactory(new PolicyTemplateExtensionManagerFactory(application, extensionModelLoaderRepository,
+                                                                                  artifactPlugins,
+                                                                                  new DefaultExtensionManagerFactory()));
+
+    artifactBuilder.withServiceConfigurator(customizationService -> customizationService
+        .overrideDefaultServiceImpl(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER,
+                                    application.getMuleContext().getRegistry()
+                                        .lookupObject(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER)));
+
+    try {
+      policyContext = artifactBuilder.build();
+      policyContext.getMuleContext().start();
+    } catch (MuleException e) {
+      throw new IllegalStateException("Cannot create artifact context for the policy instance", e);
```

Comment:
Would be better to throw an exception more related to the problem, like a lifecycle exception or similar